### DISCUSSION TITLE
Randomly crop images with specified ratio relative to the input image

### DISCUSSION THREAD

**Discussion by function2-llx** [Project-MONAI/MONAI#1950] [Ideas]
Do we need a class crops the input image with relative ratio but not some specific absolute size? If we do, I hope something like the following class makes sense:

```python
class RandSpatialCropWithRatiod(RandSpatialCropd):
    def __init__(self, keys: KeysCollection, roi_ratio: Union[Sequence[float], float], random_center: bool = True, random_size: bool = True,
                 allow_missing_keys: bool = False) -> None:
        super().__init__(keys, -1, random_center, random_size, allow_missing_keys)
        self.roi_ratio = roi_ratio

    def randomize(self, img_size: Sequence[int]) -> None:
        ratios = fall_back_tuple(self.roi_ratio, [1] * len(img_size))
        self.roi_size = [int(size * ratio) for size, ratio in zip(img_size, ratios)]
        super().randomize(img_size)

```

**Comment by Nic-Ma** [comment]
Hi @function2-llx ,

Thanks for your feature request, sounds interesting.
Do you want to contribute a PR for it?

Thanks.

**Reply by function2-llx** [reply]
Yes, I'd love to. Shall I follow [this](https://github.com/Project-MONAI/MONAI/blob/master/CONTRIBUTING.md#the-contribution-process)?

**Reply by Nic-Ma** [reply]
Yes, please check the contribution guide and make a PR.
Looking forward to it.

Thanks in advance.
